File: iloaders/import_old_flags.c
// This file is part of Adlib Tracker II (AT2).
//
// SPDX-FileType: SOURCE
// SPDX-FileCopyrightText: 2014-2026 The Adlib Tracker 2 Authors
// SPDX-License-Identifier: GPL-3.0-or-later

/*static*/ void import_old_flags (void)
{
  tFIXED_SONGDATA *song = &songdata;

  if ((song->common_flag & 0x02) != 0)
    for (uint8_t i = 0; i < 20; i++) song->lock_flags[i] |= 0x10;

  if ((song->common_flag & 0x04) != 0)
    for (uint8_t i = 0; i < 20; i++) song->lock_flags[i] |= 0x20;

  if ((song->common_flag & 0x20) != 0)
    for (uint8_t i = 0; i < 20; i++) song->lock_flags[i] &= ~0x03;
}

File: iloaders/pas/import_old_flags.pas
// This file is part of Adlib Tracker II (AT2).
//
// SPDX-FileType: SOURCE
// SPDX-FileCopyrightText: 2014-2026 The Adlib Tracker 2 Authors
// SPDX-License-Identifier: GPL-3.0-or-later

procedure import_old_flags;

var
  temp: Byte;

begin
  If (songdata.common_flag OR 2 = songdata.common_flag) then
    For temp := 1 to 20 do
      songdata.lock_flags[temp] := songdata.lock_flags[temp] OR $10;

  If (songdata.common_flag OR 4 = songdata.common_flag) then
    For temp := 1 to 20 do
      songdata.lock_flags[temp] := songdata.lock_flags[temp] OR $20;

  If (songdata.common_flag OR $20 = songdata.common_flag) then
    For temp := 1 to 20 do
      songdata.lock_flags[temp] := songdata.lock_flags[temp] AND NOT 3;
end;
